services/web-app/utils/use-query-state.js

```diff
@@ -21,58 +21,63 @@ const queryStringConfig = {
- * 3- Supplying options `parseNumbers` or `parseBoolean` = `true` will cause the type of value
+ * the hook will return the last value saved to localStorage (null if not set)
+ * 5 - By default, queryStates are always saved to localStorage;
+ * supplying option `resetOnLoad` = `false` will cause the value to be removed from localStorage
+ * everytime a new page navigation to the path occurs
+ * 6- Supplying options `parseNumbers` or `parseBoolean` = `true` will cause the type of value
  * to be casted to desired type if possible, otherwise type will be string
- * 4 - Supplying option `saveToStorage` = `true` will cause the value
- * to be stored within browser's localStorage. This value will be used instead of
- * the query string value if it happens to be invalid or undefined
- * 5 - Calling the update function with a `null`, or `undefined` value
- * will cause the localStorage entry to be removed (if any)
  * @param {string} key Key to use in the query string
- * @param {{defaultValue: string, saveToStorage: boolean,
- * parseNumbers: boolean, parseBooleans: boolean}} options Extra config options
+ * @param {{defaultValue: string, parseNumbers: boolean,
+ * parseBooleans: boolean, resetOnLoad: boolean}} options Extra config options
  *
  * @returns {{value, update}} Current value and update function
  */
 const useQueryState = (
   key,
-  { defaultValue = '', saveToStorage = false, parseNumbers = false, parseBooleans = false },
+  { defaultValue = '', parseNumbers = false, parseBooleans = false, resetOnLoad = true },
   validateValue = () => true
 ) => {
   const router = useRouter()
 
-  const getValue = useCallback(() => {
-    if (typeof window === 'undefined') {
-      return null
-    }
-
-    const query = queryString.parseUrl(window.location.search, {
-      ...queryStringConfig,
-      parseNumbers,
-      parseBooleans
-    }).query
+  const getValue = useCallback(
+    (isInternalCall = false) => {
+      if (typeof window === 'undefined') {
+        return null
+      }
 
-    const storageValue = saveToStorage
-      ? localStorage.getItem(`${router.pathname}:${key}`)
-      : undefined
+      const query = queryString.parseUrl(window.location.search, {
+        ...queryStringConfig,
+        parseNumbers,
+        parseBooleans
+      }).query
 
-    let val = query[key]
+      if (!isInternalCall && resetOnLoad) {
+        localStorage.setItem(`${router.pathname}:${key}`, defaultValue)
+      }
 
-    if (val !== undefined && typeof validateValue === 'function' && !validateValue(val)) {
-      val = undefined
-    }
+      let val = query[key]
 
-    if (val === undefined) {
-      val = storageValue
-    }
+      if (typeof validateValue === 'function' && !validateValue(val)) {
+        const storageValue = localStorage.getItem(`${router.pathname}:${key}`)
+        if (parseBooleans) {
+          val = storageValue === 'true'
+        } else if (parseNumbers) {
+          val = Number(storageValue)
```

Comment:
I'm not sure these are working as intended. On line 69, if the value coming out of local storage is `"a-z"` (or any string other than "`true`"), val will always be returned as `false` when `parseBooleans` is `true` since the string doesn't pass this equality test and isn't meant to be interpreted as a boolean. The same thing would happen on line 71 interpreting things as numbers.

Food for thought:
For a param coming out of localstorage that you want to convert to a string or a boolean:
1. get the value out of local storage
2. invoke `querystring.stringify(...)` to convert it to a url search string
3. invoke `querystring.parse(..., {parseBooleans, parseNumbers})` against that string
4. get the value back out of the returned result and it will now be cast to the appropriate type.

Something like:

```js
const storageVal = localStorage.getItem(key)
const storageSearchQuery = queryString.stringify({key: storageVal})
const parsedStorage = queryString.parse(storageSearchQuery, {parseBooleans, parseNumbers})
const parsedActual = queryString.parse(window.location.search, {parseBooleans, parseNumbers})
if (parsedActual[key] passes validation) {return parsedActual[key]}
else {return parsedStorage[key]}
```
Note that in this example, the local storage must only contain a value that has already been shown to pass the provided validation function (this is already being done in the code you have below)

Reply:
Done! did a variation of this, can see in 4a9221ec887a36f426571900246a316ec8c78963